app/kuma-dp/pkg/dataplane/accesslogs/server.go

```diff
@@ -31,41 +30,32 @@ func (s *accessLogServer) NeedLeaderElection() bool {
 func NewAccessLogServer(dataplane kumadp.Dataplane) *accessLogServer {
 	address := envoy.AccessLogSocketName(dataplane.Name, dataplane.Mesh)
 	return &accessLogServer{
-		server:  grpc.NewServer(),
 		address: address,
 	}
 }
 
 func (s *accessLogServer) Start(stop <-chan struct{}) error {
-	v3.RegisterAccessLogServer(s.server)
-
-	_, err := os.Stat(s.address)
-	if err == nil {
-		// File is accessible try to rename it to verify it is not open
-		newName := s.address + ".bak"
-		err := os.Rename(s.address, newName)
-		if err != nil {
-			return errors.Errorf("file %s exists and probably opened by another kuma-dp instance", s.address)
-		}
-		err = os.Remove(newName)
-		if err != nil {
-			return errors.Errorf("not able the delete the backup file %s", newName)
-		}
+	alStreamer := v3.NewAccessLogStreamer()
+	os.Remove(s.address)
```

Comment:
unhandled error. Why are we removing it?

Reply:
The existing stat -> mv is a classic race between checking file and using it, so I dropped that whole block. Then, it really doesn't matter if remove fails (eg if file does not exist) as long as mkfifo succeeds, and you could even imagine a race between remove and mkfifo which invalidates the success/failure response, which is an argument to not check it at all.

I can add an error check that just ignores ENOENT though.